sys/net/gnrc/transport_layer/tcp/tcp_internal/option.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2015 Simon Brummer
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_gnrc_tcp TCP
+ * @ingroup     net_gnrc
+ * @brief       RIOT's tcp implementation for the gnrc stack
+ *
+ * @{
+ *
+ * @file
+ * @brief      Defines and Macros for TCP option handling
+ *
+ * @author     Simon Brummer <brummer.simon@googlemail.com>
+ * @}
+ */
+
+
+#ifndef GNRC_TCP_INTERNAL_OPTION_H_
+#define GNRC_TCP_INTERNAL_OPTION_H_
+
+#include "helper.h"
+#include "net/conn/tcp.h"
+#include "net/gnrc/tcp/hdr.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Offset without Options(in 32-Bit words) */
+ #define OPTION_OFFSET_BASE 0x5
+ #define OPTION_OFFSET_MAX  0xF
+
+ #define GET_OFFSET( x ) (((x) & MSK_OFFSET) >> 12)
```

Comment:
Also, is there any case where you need the "compressed" value? Why not just return the actual header length in byte.